safe_transaction_service/events/services/queue_service.py

```diff
@@ -1,276 +1,151 @@
-            on_open_error_callback=self.on_connection_open_error,
-            on_close_callback=self.on_connection_closed,
-        )
+            # Send messages if there was any missing
+            # self.send_unsent_events()
+            return self.connection
+        except pika.exceptions.AMQPConnectionError:
+            logger.error("Cannot open connection with RabbitMQ")
 
-    def on_connection_open(self, connection: GeventConnection):
+    def is_connected(self) -> bool:
         """
-        This method is called by pika once the connection to RabbitMQ has
-        been established. It passes the handle to the connection object.
 
-        :param GeventConnection connection: The connection
+        :return: True if is connected False in other case
         """
+        if not self.connection or not self.connection.is_open:
+            return False
+        return True
 
-        logger.info("Connection opened with %s", self._connection_parameters.host)
-        self._connection = connection
-        self.open_channel()
-
-    def on_connection_open_error(self, connection: GeventConnection, err: Exception):
+    def publish(self, message: str) -> bool:
         """
-        This method is called by pika if the connection to RabbitMQ
-        can't be established. Connection object is paased if were necessary
-        Always retry the reconnection every 5 seconds.
 
-        :param GeventConnection: The connection
-        :param Exception err: The error
+        :param message:
+        :return: True if message was published False in other case
         """
-        logger.error(
-            "Connection open failed with %s, retrying in 5 seconds: %s",
-            self._connection_parameters.host,
-            err,
-        )
-        connection.ioloop.call_later(5, self.connect)
+        # Check if is still connected if not try to reconnect
+        if not self.is_connected() and not self.connect():
+            return False
+        try:
+            self.channel.basic_publish(
+                exchange=self.exchange_name, routing_key="", body=message
+            )
+            return True
+        except pika.exceptions.AMQPConnectionError:
+            return False
 
-    def on_connection_closed(self, connection: GeventConnection, reason: Exception):
-        """
-        This method is invoked by pika when the connection to RabbitMQ is
-        closed unexpectedly. Since it is unexpected, we will reconnect to
-        RabbitMQ if it disconnects.
 
-        :param GeventConnection: The closed connection obj
-        :param Exception reason: exception representing reason for loss of
-            connection.
-        """
-        self._channel = None
-        logger.error(
-            "Connection closed with %s, reopening in 5 seconds: %s",
-            self._connection_parameters.host,
-            reason,
-        )
-        connection.ioloop.call_later(5, self.connect)
+@lru_cache
```

Comment:
You can use @cache , as we don't need to support previous Python versions (in safe-eth-py we do)